code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/InternalMessage.java

```diff
@@ -114,12 +115,7 @@ public InternalMessage(final MessagingExtension parent, final RuleConsequence co
         final MessageSettings settings = messageSettingsFromMap(rawMessageSettings);
         settings.setParent(this);
 
-        // set the internal Messaging delegate if a custom Messaging delegate is not being used
-        if (ServiceProvider.getInstance().getMessageDelegate() == null) {
-            ServiceProvider.getInstance().setMessageDelegate(this);
-        }
-
-        aepMessage = ServiceProvider.getInstance().getUIService().createFullscreenMessage(html, ServiceProvider.getInstance().getMessageDelegate(), !assetMap.isEmpty(), settings);
+        aepMessage = ServiceProvider.getInstance().getUIService().createFullscreenMessage(html, this, !assetMap.isEmpty(), settings);
```

Comment:
should we be breaking this up to prevent against NPEs?

Reply:
yup, i can do that to make it safer.